core/src/Kokkos_GraphNode.hpp

```diff
@@ -228,6 +228,28 @@ class GraphNodeRef {
   //----------------------------------------------------------------------------
   // <editor-fold desc="then_parallel_for"> {{{2
 
+  // Helper for the 'then', such that the user can indeed pass a callable that
+  // takes no argument.
+  template <typename Functor>
+  struct Wrapper {
```

Comment:
It could also be named something like `SingleWrapper` if we'd want to have the "single" idea.

Reply:
I went for `ThenWrapper`.